xdsl/frontend/op_resolver.py

```diff
@@ -0,0 +1,69 @@
+import ast
+import importlib
+import inspect
+
+from dataclasses import dataclass
+from typing import Callable, Type
+from xdsl.frontend.dialects.builtin import _FrontendType
+from xdsl.frontend.exception import FrontendProgramException
+from xdsl.ir import Operation
+
+
+@dataclass
+class OpResolver:
+    """
+    Class responsible to resolve frontend operations into xDSL operations.
+    """
+
+    @staticmethod
+    def resolve_op(module_name: str,
+                   func_name: str) -> Callable[..., Operation]:
+        module = importlib.import_module(module_name)
```

Comment:
Importing inline always feels like crazy hacks to me. Is there a way to define this differently? Not in an extensible way, right? Maybe some form of a context would help?

I guess this is the frontend after all, so it will be hacky. I am fine with leaving as is, but I wanted to raise the point :)

Reply:
I am not aware of any other ways because we need to get the module which holds all classes and constructors. One way would be to have one module for all dialects, but this seems... not so great either.

But we can think more of it. The module path is good because it can be autogenerated easily.